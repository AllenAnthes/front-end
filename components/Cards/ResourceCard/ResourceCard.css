.ResourceCard {
  width: 400px;
  font-family: var(--secondaryFontFamily), sans-serif;
  justify-content: normal;
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  transition: 0.2s box-shadow linear;
}

.ResourceCard:focus,
.ResourceCard:hover {
  box-shadow: 0 3px 6px 2px var(--primary);
}

@media screen and (--small-viewport) {
  .ResourceCard {
    width: calc(100% - 5rem); /* subtracting margin/padding */
  }
}

.titleSection {
  display: flex;
  align-items: center;
  width: 100%;
}

.titleSection img {
  height: 50px;
}

.titleSection h5 {
  margin-left: 0.5rem;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@media screen and (--small-viewport) {
  .titleSection h5 {
    font-size: 1.25rem;
  }
}

.descriptionText {
  line-height: 1.25;
  font-weight: 200;
  overflow: hidden;
  position: relative;
  height: 6.2em; /* exactly five lines */
}

/* Text fade effect */
.descriptionText::after {
  content: "";
  text-align: right;
  position: absolute;
  bottom: 0;
  right: 0;
  width: 70%;
  height: 1.2em;
  background: linear-gradient(to right, rgba(247, 247, 247, 0), rgba(247, 247, 247, 1) 80%);
}

.footerSection {
  display: flex;
  width: 100%;
  padding-top: 1rem;
  align-items: center;
  border-top: 1px solid var(--gray);
}

.footerText {
  color: rgba(var(--rgbValuesSecondary), 0.75);
  margin-right: 1.5rem;
}

.voteBtn {
  display: flex;
  text-decoration: none;
  padding: 0.5rem 0.25rem;
  margin: 0;
  border: none;
  background: inherit;
}

.voteBtn svg {
  height: 1rem;
  margin: 0 0.5rem;
}

.voteBtn svg.active {
  color: var(--primary);
}

.voteBtn:hover {
  cursor: pointer;
}

.faded {
  opacity: 0.4;
}
