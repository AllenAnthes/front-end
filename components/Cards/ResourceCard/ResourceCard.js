import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Card from 'components/Cards/Card/Card';
import ThumbsUp from 'static/images/icons/FontAwesome/thumbs-up.svg';
import ThumbsDown from 'static/images/icons/FontAwesome/thumbs-down.svg';
import styles from './ResourceCard.css';

ResourceCard.propTypes = {
  name: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
  description: PropTypes.string,
  upvotes: PropTypes.number,
  upvoted: PropTypes.bool,
  downvotes: PropTypes.number,
  downvoted: PropTypes.bool,
  href: PropTypes.string,
  className: PropTypes.string,
};

ResourceCard.defaultProps = {
  className: undefined,
  description: '',
  upvotes: 0,
  upvoted: false,
  downvotes: 0,
  downvoted: false,
  href: '',
};

export default function ResourceCard({
  imageSource,
  name,
  description,
  upvotes,
  upvoted,
  downvotes,
  downvoted,
  className,
}) {
  return (
    <Card className={classnames(className, styles.ResourceCard)}>
      <div className={styles.titleSection}>
        <img src={imageSource} alt="logo" />
        <h5>{name}</h5>
      </div>

      <div>
        <p className={styles.descriptionText}>{description}</p>
      </div>

      <div className={styles.footerSection}>
        <span className={styles.footerText}>I found this useful </span>
        <button type="button" className={classnames(styles.voteBtn, { [styles.faded]: !upvotes })}>
          <ThumbsUp className={classnames(styles.icon, { [styles.active]: upvoted })} />
          {upvotes}
        </button>

        <button
          type="button"
          className={classnames(styles.voteBtn, { [styles.faded]: !downvotes })}
        >
          <ThumbsDown className={classnames(styles.icon, { [styles.active]: downvoted })} />

          {downvotes}
        </button>
      </div>
    </Card>
  );
}
