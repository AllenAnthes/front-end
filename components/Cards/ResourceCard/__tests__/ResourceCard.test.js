/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ResourceCard from '../ResourceCard';

// eslint-disable-next-line max-len
const longDescription = `Cliche umami truffaut scenester vice bespoke. Lomo copper mug seitan raw denim tousled coloring book kinfolk irony venmo fam austin. Pitchfork vape pork belly, fixie flannel enamel pin cold-pressed echo park. Organic sriracha waistcoat austin keytar, gentrify cardigan mumblecore PBR&B tumblr normcore meggings scenester. Enamel pin kitsch butcher, synth etsy next level gluten-free. Put a bird on it vexillologist XOXO iPhone wolf coloring book chicharrones 8-bit VHS unicorn. Leggings viral taxidermy keytar, cray schlitz hot chicken kogi truffaut art party XOXO.`;

// eslint-disable-next-line max-len
const shortDescription = `Trust fund post-ironic austin offal. Asymmetrical knausgaard selfies, butcher celiac brunch brooklyn palo santo iceland gochujang chicharrones waistcoat plaid.`;

describe('ResourceCard', () => {
  it('should render with required props', () => {
    createSnapshotTest(
      <ResourceCard
        imageSource="/static/images/icons/javascript_logo.svg"
        name="JavaScript for Dummies"
      />,
    );
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <ResourceCard
        className="test-class"
        imageSource="/static/images/icons/javascript_logo.svg"
        name="JavaScript for Dummies"
        upvotes={12}
        downvotes={25}
        description={shortDescription}
      />,
    );
  });

  it('should render properly with truncated description', () => {
    createSnapshotTest(
      <ResourceCard
        className="test-class"
        imageSource="/static/images/icons/javascript_logo.svg"
        name="JavaScript for Super Duper Dummies"
        upvotes={12}
        downvotes={25}
        description={longDescription}
      />,
    );
  });

  it('should render properly with faded up/downvotes', () => {
    createSnapshotTest(
      <ResourceCard
        className="test-class"
        imageSource="/static/images/icons/javascript_logo.svg"
        name="JavaScript for Dummies"
        upvotes={0}
        downvotes={0}
        description={shortDescription}
      />,
    );
  });
});
