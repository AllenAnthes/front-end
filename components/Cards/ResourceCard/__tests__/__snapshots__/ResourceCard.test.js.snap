// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCard should render properly with faded up/downvotes 1`] = `
<article
  className="Card test-class ResourceCard"
>
  <div
    className="titleSection"
  >
    <img
      alt="logo"
      src="/static/images/icons/javascript_logo.svg"
    />
    <h5>
      JavaScript for Dummies
    </h5>
  </div>
  <div>
    <p
      className="descriptionText"
    >
      Trust fund post-ironic austin offal. Asymmetrical knausgaard selfies, butcher celiac brunch brooklyn palo santo iceland gochujang chicharrones waistcoat plaid.
    </p>
  </div>
  <div
    className="footerSection"
  >
    <span
      className="footerText"
    >
      I found this useful 
    </span>
    <button
      className="voteBtn faded"
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
    <button
      className="voteBtn faded"
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
  </div>
</article>
`;

exports[`ResourceCard should render properly with truncated description 1`] = `
<article
  className="Card test-class ResourceCard"
>
  <div
    className="titleSection"
  >
    <img
      alt="logo"
      src="/static/images/icons/javascript_logo.svg"
    />
    <h5>
      JavaScript for Super Duper Dummies
    </h5>
  </div>
  <div>
    <p
      className="descriptionText"
    >
      Cliche umami truffaut scenester vice bespoke. Lomo copper mug seitan raw denim tousled coloring book kinfolk irony venmo fam austin. Pitchfork vape pork belly, fixie flannel enamel pin cold-pressed echo park. Organic sriracha waistcoat austin keytar, gentrify cardigan mumblecore PBR&B tumblr normcore meggings scenester. Enamel pin kitsch butcher, synth etsy next level gluten-free. Put a bird on it vexillologist XOXO iPhone wolf coloring book chicharrones 8-bit VHS unicorn. Leggings viral taxidermy keytar, cray schlitz hot chicken kogi truffaut art party XOXO.
    </p>
  </div>
  <div
    className="footerSection"
  >
    <span
      className="footerText"
    >
      I found this useful 
    </span>
    <button
      className="voteBtn"
      type="button"
    >
      <svg
        className="icon"
      />
      12
    </button>
    <button
      className="voteBtn"
      type="button"
    >
      <svg
        className="icon"
      />
      25
    </button>
  </div>
</article>
`;

exports[`ResourceCard should render with many props assigned 1`] = `
<article
  className="Card test-class ResourceCard"
>
  <div
    className="titleSection"
  >
    <img
      alt="logo"
      src="/static/images/icons/javascript_logo.svg"
    />
    <h5>
      JavaScript for Dummies
    </h5>
  </div>
  <div>
    <p
      className="descriptionText"
    >
      Trust fund post-ironic austin offal. Asymmetrical knausgaard selfies, butcher celiac brunch brooklyn palo santo iceland gochujang chicharrones waistcoat plaid.
    </p>
  </div>
  <div
    className="footerSection"
  >
    <span
      className="footerText"
    >
      I found this useful 
    </span>
    <button
      className="voteBtn"
      type="button"
    >
      <svg
        className="icon"
      />
      12
    </button>
    <button
      className="voteBtn"
      type="button"
    >
      <svg
        className="icon"
      />
      25
    </button>
  </div>
</article>
`;

exports[`ResourceCard should render with required props 1`] = `
<article
  className="Card ResourceCard"
>
  <div
    className="titleSection"
  >
    <img
      alt="logo"
      src="/static/images/icons/javascript_logo.svg"
    />
    <h5>
      JavaScript for Dummies
    </h5>
  </div>
  <div>
    <p
      className="descriptionText"
    >
      
    </p>
  </div>
  <div
    className="footerSection"
  >
    <span
      className="footerText"
    >
      I found this useful 
    </span>
    <button
      className="voteBtn faded"
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
    <button
      className="voteBtn faded"
      type="button"
    >
      <svg
        className="icon"
      />
      0
    </button>
  </div>
</article>
`;
